src/map.cpp

```diff
@@ -1430,33 +1446,44 @@ void Map::timerUpdate(float dtime, float unload_timeout,
 			MapBlock *block = (*i);
 
 			block->incrementUsageTimer(dtime);
+			mapblock_queue.push(TimeOrderedMapBlock(sector, block));
```

Comment:
1. `The Map::timerUpdate` function is run every [5.2 seconds](https://github.com/minetest/minetest/blob/master/src/client.cpp#L419), not every step (in server too)
2. The "current order" isn't based on creation date, but on used date. This date is updated on the client in the [ClientMap::updateDrawList](https://github.com/minetest/minetest/blob/master/src/clientmap.cpp#L324) function, which is called [every 200 ms](https://github.com/minetest/minetest/blob/master/src/game.cpp#L3974). For the server, this happens at every environment step.

You'd need to remove and re-insert the mapblock from the data structure at every update. While this would affect fewer mapblocks as for rebuilding the structure in `timerUpdate`, its more often neccessary. Also, it would require a new data structure.